ec-divisors-contest/src/lib.rs

```diff
@@ -29,14 +29,14 @@ pub fn check_init_contest(point: &EdwardsPoint, scalar: &Scalar) {
     let (_, _) = <EdwardsPoint as DivisorCurve>::to_xy(*point).expect("zero scalar was decomposed");
 }
 
-pub fn run_bench_ref(point: &EdwardsPoint, scalar: &Scalar) -> PolyRef<FieldElement> {
+pub fn run_bench_ref(point: &EdwardsPoint, scalar: &Scalar) -> (PolyRef<FieldElement>, ScalarDecompositionRef<Scalar>) {
     let scalar = ScalarDecompositionRef::new(*scalar).unwrap();
-    scalar.scalar_mul_divisor(*point)
+    (scalar.scalar_mul_divisor(*point), scalar)
 }
 
-pub fn run_bench_contest(point: &EdwardsPoint, scalar: &Scalar) -> Poly<FieldElement> {
+pub fn run_bench_contest(point: &EdwardsPoint, scalar: &Scalar) -> (Poly<FieldElement>, ScalarDecomposition<Scalar>) {
     let scalar = ScalarDecomposition::new(*scalar).unwrap();
-    scalar.scalar_mul_divisor(*point)
+    (scalar.scalar_mul_divisor(*point), scalar)
 }
```

Comment:
`ScalarDecomposition` contains a `Vec` which got dropped in this function and included in the benchmark. This returns it to be ignored by `iter_with_large_drop`.